File: src/test/java/com/raychenon/leetcode/linkedlist/PalindromeLinkedListTest.java
package com.raychenon.leetcode.linkedlist;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.Arrays;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-11-10
 * https://leetcode.com/problems/palindrome-linked-list/
 */
public class PalindromeLinkedListTest {

    private static PalindromeLinkedList instance;

    @BeforeClass
    public static void setUp() {
        instance = new PalindromeLinkedList();
    }

    @Test
    public void palindromeLinkedListTest() {
        ListNode root = ListNode.createList(Arrays.asList(1, 2, 2, 1));
        assertEquals(true, instance.isPalindrome(root));
    }

    @Test
    public void palindromeSizeOddTest() {
        ListNode root = ListNode.createList(Arrays.asList(1, 2, 3, 2, 1));
        assertEquals(true, instance.isPalindrome(root));
    }

    @Test
    public void notPalindromeTest() {
        ListNode root = ListNode.createList(Arrays.asList(1, 2));
        assertEquals(false, instance.isPalindrome(root));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/linkedlist/PalindromeLinkedListTest.kt
package com.raychenon.kotlin.leetcode.linkedlist

import com.raychenon.leetcode.linkedlist.ListNode
import junit.framework.Assert.assertEquals
import org.junit.Test
import java.util.*

/**
 * User: raychenon
 * Date: 2019-11-10
 * https://leetcode.com/problems/palindrome-linked-list
 */
class PalindromeLinkedListTest {

    @Test
    fun palindromeLinkedListTest() {
        val root = ListNode.createList(Arrays.asList(1, 2, 2, 1))
        assertEquals(true, PalindromeLinkedList.isPalindrome(root))
    }

    @Test
    fun palindromeSizeOddTest() {
        val root = ListNode.createList(Arrays.asList(1, 2, 3, 2, 1))
        assertEquals(true, PalindromeLinkedList.isPalindrome(root))
    }

    @Test
    fun notPalindromeTest() {
        val root = ListNode.createList(Arrays.asList(1, 2))
        assertEquals(false, PalindromeLinkedList.isPalindrome(root))
    }

}